app/script/util/util.js

```diff
@@ -232,28 +232,40 @@ z.util.base64ToBlob = base64 => {
 /**
  * Downloads blob using a hidden link element.
  * @param {Blob} blob - Blob to store
- * @param {string} filename - Data will be saved under this name
+ * @param {string} fileName - Data will be saved under this name
  * @param {string} [mimeType] - Mime type of the generated download
  * @returns {number} Timeout identifier
  */
 
-z.util.downloadBlob = (blob, filename, mimeType) => {
+z.util.downloadBlob = (blob, fileName, mimeType) => {
   const url = window.URL.createObjectURL(blob);
-  const link = document.createElement('a');
-  // firefox needs the element to be in the DOM for the download to start
-  // see https://stackoverflow.com/a/32226068
-  document.body.appendChild(link);
-  link.href = url;
-  link.download = filename;
-  link.style = 'display: none';
+  const anchor = document.createElement('a');
+  anchor.href = url;
   if (mimeType) {
-    link.type = mimeType;
+    anchor.type = mimeType;
   }
-  link.click();
+  return z.util.downloadFile(anchor, url, fileName);
```

Comment:
I thin the download file method should take the href content, name and an optional mimetype and create the link element itself. Looks like this would simplify the code.

Reply:
Ok, I changed that.